intel_extension_for_transformers/llm/runtime/graph/cmake/Common.cmake

```diff
@@ -36,9 +36,28 @@ function(add_executable_w_warning TARGET)
     warning_check(${TARGET})
 endfunction()
 
-function(add_library_w_warning TARGET)
-    add_library(${TARGET} STATIC ${ARGN})
+function(add_library_w_warning_ TARGET)
+    add_library(${TARGET} ${ARGN})
     set_target_properties(${TARGET} PROPERTIES C_STANDARD 11 C_STANDARD_REQUIRED ON C_EXTENSIONS OFF)
     set_target_properties(${TARGET} PROPERTIES CXX_STANDARD 11 CXX_STANDARD_REQUIRED ON CXX_EXTENSIONS OFF)
     warning_check(${TARGET})
 endfunction()
+
+function(add_library_w_warning TARGET)
+    add_library_w_warning_(${TARGET} STATIC ${ARGN})
+endfunction()
+
+function(add_shared_library_w_warning TARGET)
+    add_library_w_warning_(${TARGET} SHARED ${ARGN})
+endfunction()
+
+function(add_shareable_library_w_warning TARGET)
+    if (BUILD_SHARED_LIBS)
+    add_library_w_warning_(${TARGET} SHARED ${ARGN})
```

Comment:
format spaces?

Reply:
Ther is no formal format for cmake? Can be changed in next PR